File: backend/app/schemas/setor.py
"""
Schemas de validação de entrada/saída do catálogo de Setores.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class SetorCreate(BaseModel):
    nome: str = Field(..., min_length=2, max_length=100)
    descricao: str | None = Field(default=None, max_length=300)


class SetorUpdate(BaseModel):
    nome: str | None = Field(default=None, min_length=2, max_length=100)
    descricao: str | None = Field(default=None, max_length=300)


class SetorOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nome: str
    descricao: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime


class SetorListOut(BaseModel):
    total: int
    items: list[SetorOut]

File: frontend/src/types/setor.ts
export interface Setor {
  id: string;
  nome: string;
  descricao: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SetorListagem {
  total: number;
  items: Setor[];
}

export interface SetorFormData {
  nome: string;
  descricao?: string | null;
}
